packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/base-types.ts

```diff
@@ -826,6 +826,37 @@ export class AcceleratorType {
   }
 }
 
+/**
+ * Specifies the data to be used for inference,
+ * and to associate the data that is relevant to the prediction results in the output.
+ *
+ * @see https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_DataProcessing.html
+ * @experimental
+ */
+export interface DataProcessing {
+  /**
+   * JSONPath expression used to select a portion of the input data to pass to the algorithm.
+   *
+   * @default - $
+   */
+  readonly inputFilter?: string;
+
+  /**
+   * Specifies the source of the data to join with the transformed data.
+   *
+   * @default - None
+   */
+  readonly joinSource?: 'Input' | 'None';
```

Comment:
this will need to be strongly typed as it won't translate to other languages.
it could be an enum or an enum like class depending on what's appropriate